show_sale/models/sale_order.py

```diff
@@ -17,8 +19,23 @@ class SaleOrder(models.Model):
 
     @api.onchange("show_place_id")
     def onchange_show_place(self):
-        self.show_place_maximum_capacity = self.show_place_id.show_place_maximum_capacity
+        self.show_place_maximum_capacity = (
+            self.show_place_id.show_place_maximum_capacity
+        )
 
     @api.onchange("show_project_id")
     def onchange_show_project(self):
         self.analytic_account_id = self.show_project_id.analytic_account_id
+
+    @api.multi
+    @api.constrains("show_project_id")
+    def _check_unique_so(self):
+        for so in self:
+            if so.show_project_id and self.search(
```

Comment:
Tu peux faire simplement:
if so.show_project_id and if so.show_project_id.show_sale_order_count > 0:

Reply:
Je garde la fonction telle qu'elle est , car on doit checker si id de sale order est different de so actuel.
("id", "!=", so.id)